src/targets.rs

```diff
@@ -1685,6 +1694,9 @@ mod tests {
             "wasm32-unknown-emscripten",
             "wasm32-unknown-unknown",
             "wasm32-wasi",
+            "wasm32-wasip1-threads",
```

Comment:
wasm32-wasi-preview1-threads hasn't been renamed to wasm32-wasip1-threads yet and there is no open PR for this either.

Reply:
Indeed! I was a bit forward thinking adding this in, so I suspect it won't be used for a bit.

Reply:
Once the wasip{1,2} targets land in rustc I'll post a follow-up to rename preview1-threads to p1-threads, but in the meantime it seems reasonable to add the target here too